packages/ui/src/models/camel/camel-k-resource.ts

```diff
@@ -94,12 +100,15 @@ export abstract class CamelKResource implements CamelResource {
   getCompatibleComponents(_mode: AddStepMode, _visualEntityData: IVisualizationNodeData): TileFilter | undefined {
     return undefined;
   }
+  getSerializer() {
+    return this.serializer;
+  }
```

Comment:
Do we need external access to the serializer? I have the feeling that we don't, so probably we can remove this entirely.
```suggestion
```